packages/channel-provider/src/channel-provider.ts

```diff
@@ -76,6 +87,12 @@ class ChannelProvider implements ChannelProviderInterface {
 
   off: OffType = (method, params) => this.events.off(method, params);
 
+  private async populateProviderProperties() {
+    this.internalAddress = await this.send({method: 'GetAddress', params: {}});
+    this.selectedAddress = await this.send({method: 'GetEthereumSelectedAddress', params: {}});
+    this.walletVersion = await this.send({method: 'WalletVersion', params: {}});
+  }
```

Comment:
I like this. It seems to hint to us to combine these RPC calls into one at some point in the future. 

Should we make the `WalletVersion` check *first*? If that fails, we should get a meaningful error and not bother with the other calls.

Reply:
I think there is a separate thing we can do which is to determine whether or not the iFrame is in fact loaded and the PostMessage bridge is behaving correctly. Ideally there would be some `testConnection` method or something like that. We could have a property on `ChannelProvider` which is `connectionEstablished: boolean` which would be set to `true` on an event monitoring the frame [like this one](https://stackoverflow.com/a/36155560/2680092) being triggered.